packages/nc-gui/components/smartsheet/Kanban.vue

```diff
@@ -0,0 +1,493 @@
+  IsGalleryInj,
+  IsGridInj,
+  IsKanbanInj,
+  IsLockedInj,
+  IsPublicInj,
+  MetaInj,
+  OpenNewRecordFormHookInj,
+  ReadonlyInj,
+  inject,
+  onMounted,
+  provide,
+  useKanbanViewData,
+} from '#imports'
+import type { Row as RowType } from '~/composables'
+
+const meta = inject(MetaInj, ref())
+
+const view = inject(ActiveViewInj, ref())
+
+const reloadViewDataHook = inject(ReloadViewDataHookInj)
+
+const reloadViewMetaHook = inject(ReloadViewMetaHookInj)
+
+const openNewRecordFormHook = inject(OpenNewRecordFormHookInj, createEventHook())
+
+const isLocked = inject(IsLockedInj, ref(false))
+
+const isPublic = inject(IsPublicInj, ref(false))
+
+const expandedFormDlg = ref(false)
+
+const expandedFormRow = ref<RowType>()
+
+const expandedFormRowState = ref<Record<string, any>>()
+
+const deleteStackVModel = ref(false)
+
+const stackToBeDeleted = ref('')
+
+const stackIdxToBeDeleted = ref(0)
+
+const route = useRoute()
+
+const router = useRouter()
+
+const {
+  loadKanbanData,
+  loadMoreKanbanData,
+  loadKanbanMeta,
+  kanbanMetaData,
+  formattedData,
+  updateOrSaveRow,
+  updateKanbanMeta,
+  addEmptyRow,
+  groupingFieldColOptions,
+  updateKanbanStackMeta,
+  groupingField,
+  countByStack,
+  deleteStack,
+  removeRowFromUncategorizedStack,
+  shouldScrollToRight,
+} = useKanbanViewData(meta, view)
+
+const { isUIAllowed } = useUIPermission()
+
+const { appInfo } = $(useGlobal())
+
+provide(IsFormInj, ref(false))
+
+provide(IsGalleryInj, ref(false))
+
+provide(IsGridInj, ref(false))
+
+provide(IsKanbanInj, ref(true))
+
+provide(ReadonlyInj, !isUIAllowed('xcDatatableEditable'))
+
+const fields = inject(FieldsInj, ref([]))
+
+const kanbanContainerRef = ref()
+
+let isMounted = false
+
+const isRowEmpty = (record: any, col: any) => {
+  const val = record.row[col.title]
+  if (!val) return true
+
+  return Array.isArray(val) && val.length === 0
+}
+
+reloadViewDataHook?.on(async () => {
+  await loadKanbanMeta()
+  await loadKanbanData()
+})
+
+reloadViewMetaHook?.on(async () => {
+  await loadKanbanMeta()
+})
+
+const expandForm = (row: RowType, state?: Record<string, any>) => {
+  if (!isUIAllowed('xcDatatableEditable')) return
+
+  const rowId = extractPkFromRow(row.row, meta.value!.columns!)
+
+  if (rowId) {
+    router.push({
+      query: {
+        ...route.query,
+        rowId,
+      },
+    })
+  } else {
+    expandedFormRow.value = row
+    expandedFormRowState.value = state
+    expandedFormDlg.value = true
+  }
+}
+
+const expandedFormOnRowIdDlg = computed({
+  get() {
+    return !!route.query.rowId
+  },
+  set(val) {
+    if (!val)
+      router.push({
+        query: {
+          ...route.query,
+          rowId: undefined,
+        },
+      })
+  },
+})
+
+const expandFormClick = async (e: MouseEvent, row: RowType) => {
+  if (e.target as HTMLElement) {
+    expandForm(row)
+  }
+}
+
+/** Block dragging the stack to first index (reserved for uncategorized) **/
+function onMoveCallback(event: any) {
```

Comment:
Is there no type for this event? All of these are `any`.
Can you provide an interface?